openedx/core/djangoapps/verified_track_content/partition_scheme.py

```diff
@@ -0,0 +1,138 @@
+        ]
+
+    def to_json(self):
+        """
+        Because this partition is dynamic, to_json and from_json are not supported.
+
+        Calling this method will raise a TypeError.
+        """
+        raise TypeError("Because EnrollmentTrackUserPartition is a dynamic partition, 'to_json' is not supported.")
+
+    def from_json(self):
+        """
+        Because this partition is dynamic, to_json and from_json are not supported.
+
+        Calling this method will raise a TypeError.
+        """
+        raise TypeError("Because EnrollmentTrackUserPartition is a dynamic partition, 'from_json' is not supported.")
+
+
+class EnrollmentTrackPartitionScheme(object):
+    """
+    This scheme uses learner enrollment tracks to map learners into partition groups.
+    """
+
+    @classmethod
+    def get_group_for_user(cls, course_key, user, user_partition, **kwargs):  # pylint: disable=unused-argument
+        """
+        Returns the Group from the specified user partition to which the user
+        is assigned, via enrollment mode.
+
+        If a course is using the Verified Track Cohorting pilot feature, this method
+        returns None regardless of the user's enrollment mode.
+        """
+        if is_course_using_cohort_instead(course_key):
+            return None
+
+        # NOTE: masquerade code was copied from CohortPartitionScheme, and it may need
+        # some changes (or if not, code should be refactored out and shared).
+        # This work will be done in a future story TNL-6739.
+
+        # First, check if we have to deal with masquerading.
+        # If the current user is masquerading as a specific student, use the
+        # same logic as normal to return that student's group. If the current
+        # user is masquerading as a generic student in a specific group, then
+        # return that group.
+        if get_course_masquerade(user, course_key) and not is_masquerading_as_specific_student(user, course_key):
+            group_id, user_partition_id = get_masquerading_group_info(user, course_key)
+            if user_partition_id == user_partition.id and group_id is not None:
+                try:
+                    return user_partition.get_group(group_id)
+                except NoSuchUserPartitionGroupError:
+                    return None
+            # The user is masquerading as a generic student. We can't show any particular group.
+            return None
+
+        mode_slug, is_active = CourseEnrollment.enrollment_mode_for_user(user, course_key)
+        if mode_slug and is_active:
+            course_mode = CourseMode.mode_for_course(
+                course_key,
+                mode_slug,
+                modes=CourseMode.modes_for_course(course_key, include_expired=True, only_selectable=False)
+            )
+            if not course_mode:
+                course_mode = CourseMode.DEFAULT_MODE
```

Comment:
Shouldn't the logic of returning the default mode be included within `CourseMode.mode_for_course`?